src/Chainweb/Pact/Backend/Utils.hs

```diff
@@ -334,13 +327,28 @@ openSQLiteConnection file ps = do
 closeSQLiteConnection :: SQLiteEnv -> IO ()
 closeSQLiteConnection c = void $ close_v2 $ _sConn c
 
+-- passing the empty string as filename causes sqlite to use a temporary file
+-- that is deleted when the connection is closed. In practice, unless the database becomes
+-- very large, the database will reside memory and no data will be written to disk.
+--
+-- Cf. https://www.sqlite.org/inmemorydb.html
+--
 withTempSQLiteConnection :: [Pragma] -> (SQLiteEnv -> IO c) -> IO c
-withTempSQLiteConnection ps action =
-  withSystemTempFile "sqlite-tmp" (\file _ -> withSQLiteConnection file ps False action)
+withTempSQLiteConnection = withSQLiteConnection ""
```

Comment:
passing "" as database filename to sqlite, causes sqlite to create a temporary database file that is deleted when the connection is closed.